chef/cookbooks/provisioner/libraries/repositories.rb

```diff
@@ -92,40 +92,41 @@ def inspect_repos(node)
             suse_optional_repos(version, :cloud).each do |name|
               repos[name] ||= Mash.new
               next unless repos[name][:url].nil?
-              missing_cloud ||= !(File.exists?("#{node[:provisioner][:root]}/suse-#{version}/repos/#{name}/repodata/repomd.xml") ||
-                                  File.exists?("#{node[:provisioner][:root]}/suse-#{version}/repos/#{name}/suse/repodata/repomd.xml"))
+              install_cloud ||= File.exist?("#{node[:provisioner][:root]}/suse-#{version}/repos/#{name}/repodata/repomd.xml") ||
+                                File.exist?("#{node[:provisioner][:root]}/suse-#{version}/repos/#{name}/suse/repodata/repomd.xml")
             end
 
             # For pacemaker
             suse_optional_repos(version, :hae).each do |name|
               repos[name] ||= Mash.new
               next unless repos[name][:url].nil?
-              missing_hae ||= !(File.exists? "#{node[:provisioner][:root]}/suse-#{version}/repos/#{name}/repodata/repomd.xml")
+              install_hae ||= File.exist?("#{node[:provisioner][:root]}/suse-#{version}/repos/#{name}/repodata/repomd.xml")
             end
 
             # For suse storage
             suse_optional_repos(version, :storage).each do |name|
               repos[name] ||= Mash.new
               next unless repos[name][:url].nil?
-              missing_storage ||= !(File.exists? "#{node[:provisioner][:root]}/suse-#{version}/repos/#{name}/repodata/repomd.xml")
+              install_storage ||= File.exist?("#{node[:provisioner][:root]}/suse-#{version}/repos/#{name}/repodata/repomd.xml")
             end
           end
 
-          # set an attribute about missing repos so that cookbooks and crowbar
-          # know that HA cannot be used
-          # know that SUSE_Storage cannot be used
+          # set an attribute about installed repos so that cookbooks and crowbar
+          # know that HA can be used
+          # know that SUSE_Storage can be used
+          # know that OpenStack can be used
           node_set = false
           node.set[:provisioner][:suse] ||= {}
-          if node[:provisioner][:suse][:missing_cloud] != missing_cloud
-            node.set[:provisioner][:suse][:missing_cloud] = missing_cloud
```

Comment:
No, this is nothing we want to migrate because these values are set on the fly to detect if we are missing some repositories.